.github/workflows/tests.yml

```diff
@@ -0,0 +1,152 @@
+      fail-fast: false
+      matrix:
+        include:
+          # 2015
+          - python: 3.5
+            geos: 3.5.2
+            numpy: 1.10.4
+            speedups: 1
+          - python: 3.5
+            geos: 3.5.2
+            numpy: 1.10.4
+            speedups: 0
+          - python: 3.5
+            geos: 3.5.2
+            speedups: 0
+          # 2017
+          - python: 3.6
+            geos: 3.6.4
+            numpy: 1.13.3
+            speeedups: 1
+          - python: 3.6
+            geos: 3.6.4
+            numpy: 1.13.3
+            speeedups: 0
+          - python: 3.6
+            geos: 3.6.4
+            speeedups: 0
+          # 2018
+          - python: 3.7
+            geos: 3.7.3
+            numpy: 1.15.4
+            speedups: 1
+          - python: 3.7
+            geos: 3.7.3
+            numpy: 1.15.4
+            speedups: 0
+          - python: 3.7
+            geos: 3.7.3
+            speedups: 0
+          # 2019
+          - python: 3.8
+            geos: 3.8.1
+            numpy: 1.17.5
+            speedups: 1
+          - python: 3.8
+            geos: 3.8.1
+            numpy: 1.17.5
+            speedups: 0
+          - python: 3.8
+            geos: 3.8.1
+            speedups: 0
+          # 2020
+          - python: 3.9
+            geos: 3.9.0
+            numpy: 1.19.5
+            speedups: 1
+          - python: 3.9
+            geos: 3.9.0
+            numpy: 1.19.5
+            speedups: 0
+          - python: 3.9
+            geos: 3.9.0
+            speedups: 0
+          # dev
+          - python: 3.9
+            geos: master
+            speedups: 1
+
+    env:
+      GEOS_VERSION: ${{ matrix.geos }}
+      GEOS_INSTALL: ${{ github.workspace }}/geosinstall/geos-${{ matrix.geos }}
+      SPEEDUPS: ${{ matrix.speedups }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Cache GEOS and pip packages
+        uses: actions/cache@v2
+        with:
+          key: ${{ runner.os }}-geos-${{ matrix.geos }}
+          path: |
+            ~/.cache/pip
+            ${{ github.workspace }}/geosinstall
+
+      - name: Install GEOS
+        shell: bash
+        run: |
+          ./ci/install_geos.sh
+          cd ${{ github.workspace }}
+
+      - name: Install python dependencies
+        shell: bash
+        run: |
+          pip install --disable-pip-version-check --upgrade pip
+          pip install --upgrade wheel
+          if [ "$GEOS_VERSION" = "master" ]; then
+            pip install --upgrade --pre Cython numpy;
+          elif [ "$SPEEDUPS" == "1" ]; then
+            pip install --install-option="--no-cython-compile" cython;
+            pip install --upgrade numpy==${{ matrix.numpy }};
+          elif [ -n "${{ matrix.numpy }}" ]; then
+            pip install --upgrade numpy==${{ matrix.numpy }};
+          fi
```

Comment:
I don't think that is needed, because numpy will never have  been installed in that case. For speedups=1, numpy is always installed, so it's only when speedups=0 that numpy needs to be installed if requested, and for the final case (no numpy) nothing additional needs to happen. 

See eg the output of `pip list` at https://github.com/jorisvandenbossche/Shapely/runs/1768150581?check_suite_focus=true#step:6:49, which is a build without numpy and confirms that numpy is not installed at that point.

Now, this led me looking at the test output to verify that indeed the numpy tests are skipped. But this is not actually the case .. Because the `pip install -v -e .[all]` we do below to install Shapely will still install numpy. 
The same happened in Travis (see eg https://travis-ci.com/github/Toblerity/Shapely/jobs/445496262), so we were not actually testing in an environment without numpy. 

So I can think we can remove the `[all]` to avoid numpy being actually installed. And I will also move the `pip list` to after the shapely install, to ensure we can inspect the environment at that point as well.

Reply:
OK, with that commit, the numpy tests are now actually properly skipped: https://github.com/jorisvandenbossche/Shapely/runs/1790452676?check_suite_focus=true

Reply:
And added a `else pip uninstall numpy;` just to be fully sure (although it didn't seem needed)